vmdb/app/models/mixins/retirement_mixin.rb

```diff
@@ -188,4 +187,18 @@ def retiring?
   def error_retiring?
     retirement_state == ERROR_RETIRING
   end
+
+  private
+
+  def setup_event_hash(requester)
+    event_hash = {:retirement_initiator => "system"}
+    event_hash[retirement_base_model_name.underscore.to_sym] = self
+    event_hash[:host] = host if self.respond_to?(:host)
+    if requester
+      event_hash[:user_id] = requester
+      event_hash[:retirement_initiator] = "user"
+    end
+    event_hash[:type] ||= self.class.name
+    event_hash
```

Comment:
@mkanoor It does not look like this ends up on the queue.  The retire_now call was queued when called from the UI or scheduler through the invoke_tasks_local method.  Then the hash is passed to `MiqAeEvent.raise_evm_event` which calls `MiqAeEngine.deliver` directly.